qubesmanager/qube_manager.py

```diff
@@ -794,6 +799,57 @@ def __init__(self, qt_app, qubes_app, dispatcher, _parent=None):
 
         self.check_updates()
 
+    def change_template(self, template):
+        selected_vms = self.get_selected_vms()
+        reply = QMessageBox.question(
+            self, self.tr("Template Change Confirmation"),
+            self.tr("Do you want to change '{0}'<br>"
+                "to Template <b>'{1}'</b>?").format(
+                ', '.join(vm.name for vm in selected_vms), template),
+            QMessageBox.Yes | QMessageBox.Cancel)
+
+        if reply == QMessageBox.Yes:
+            for info in selected_vms:
+                info.vm.template = template
```

Comment:
Setting template may fail (for example when VM is running, or when there is some issue with the template) - you need to handle errors here. The current state is the first error will interrupt the operation and leave other VMs unchanged. This may be ok if properly reported, but another option would be to collect errors for each VM individually and display summary at the end (if applicable).